livekit-agents/livekit/agents/voice/agent_session.py

```diff
@@ -363,6 +367,21 @@ def update_agent(self, agent: Agent) -> None:
                 self._update_activity_task(self._agent), name="_update_activity_task"
             )
 
+    def _create_session_close_task(
+        self, component: llm.LLM | stt.STT | tts.TTS, error: Error
+    ) -> None:
+        self.emit("session_close", SessionCloseEvent(error=error, component=component))
+        self._closing_task = asyncio.create_task(self._drain_and_close_session())
+
+    async def _drain_and_close_session(self) -> None:
+        """Close the session after waiting for any current activity to drain."""
+        # Drain the current activity if it exists
+        if self._activity is not None:
+            await self._activity.drain()
+
+        # Finally close the session
+        await self.aclose()
+
```

Comment:
I was thinking about adding the drain() method to AgentSession directly, it can be a public method.

Impl is just:
```
        if self._activity is not None:
            await self._activity.drain()
```